pkg/cache/nodes.go

```diff
@@ -212,6 +213,15 @@ func (nc *schedulerNodes) updateNode(oldNode, newNode *v1.Node) {
 		return
 	}
 
+	if schedulerNode, ok := nc.nodesMap[newNode.Name]; ok {
+		schedulerNode.ready = common.HasReadyCondition(newNode)
```

Comment:
Updating this field in `SchedulerNode` is probably not necessary, but for the sake of consistency, I decided to do it anyway.

Reply:
The `nodesMap` looks broken. I think we have a bigger issue in this code.

Reply:
I filed YUNIKORN-1091 for the broken nodesMap update.
We do need this update in the node. This change will clash with that fix.
I also logged YUNIKORN-1090 to remove the common.Node object. That change is impacted by what we have here too.
Moving forward and leaving the fix up for the two new jiras.

Reply:
I filed YUNIKORN-1091 for the broken nodesMap update.
We do need this update in the node. This change will clash with that fix.
I also logged YUNIKORN-1090 to remove the common.Node object. That change is impacted by what we have here too.
Moving forward and leaving the fix up for the two new jiras.